Foundation/URLSession/Message.swift

```diff
@@ -49,24 +49,24 @@ extension _NativeProtocol._ParsedResponseHeader {
     /// that ending.
     /// - Returns: Returning nil indicates failure. Otherwise returns a new
     ///     `ParsedResponseHeader` with the given line added.
-    func byAppending(headerLine data: Data) -> _NativeProtocol._ParsedResponseHeader? {
+    func byAppending(headerLine data: Data, headerCompleted: (String) -> Bool) -> _NativeProtocol._ParsedResponseHeader? {
         // The buffer must end in CRLF
         guard 2 <= data.count &&
             data[data.endIndex - 2] == _Delimiters.CR &&
             data[data.endIndex - 1] == _Delimiters.LF
             else { return nil }
         let lineBuffer = data.subdata(in: Range(data.startIndex..<data.endIndex-2))
         guard let line = String(data: lineBuffer, encoding: String.Encoding.utf8) else { return nil}
-        return byAppending(headerLine: line)
+        return byAppending(headerLine: line, headerCompleted: headerCompleted)
```

Comment:
I'm surprised and delighted that this happens to call the correct (private) overload, but given that the two methods have the same name and differ only by optionality of the return type, these is very confusing to read.

Please rename the private method using a leading underscore (that is, call it `_byAppending`) and call that here. It would be much clearer to the reader.